src/assisted_test_infra/test_infra/helper_classes/cluster.py

```diff
@@ -1022,8 +1022,9 @@ def wait_for_event(self, event_to_find, reference_time, params_list=None, host_i
             PendingDeprecationWarning,
         )
         handler = EventsHandler(self.api_client)
+        cluster_id = "" if infra_env_id else self.id
```

Comment:
Why?

Reply:
cause otherwise both cluster id && infra env are sent - the results are only events with both cluster id and infra env id. there are events with only infra env id

Reply:
This is deprecated function.
Use `handler.get_events(host_id, self.id, infra_env_id)` in your test instead

Reply:
No need to use this method anymore. We stopped maintaining this code.
You can create `EventsHandler` or to get by calling `events_handler` fixture.
Next, instead of calling `cluster.wait_for_event` (while ignoring the cluster id) just call `events_handler.wait_for_event` with your desired keys (`cluster_id` or `infra_env_id`)

Reply:
using fixture in test: https://gitlab.cee.redhat.com/ocp-edge-qe/kni-assisted-installer-auto/-/merge_requests/689/diffs